prometheus/internalmetrics/internal_metrics_test.go

```diff
@@ -29,7 +29,7 @@ func TestSuccessOrFailureMetric(t *testing.T) {
 
 	// simulate a failure - our failure counter metric should increment to 1
 	doSomeWork(true)
-	metrics, err = registry.Gather()
+	metrics, _ = registry.Gather()
```

Comment:
Check error here?  May not be necessay in test.

Reply:
Probably not neccesary.
I added assertions to be on the safe side.